ha/k8s_setup/ha_start.py

```diff
@@ -26,44 +29,65 @@ def usage(prog: str):
             f"where:\n"
-            f"--services health_monitor, fault_tolerance, k8s_monitor\n"
-            f"--config   Config URL\n")
+            f"-s --services health_monitor, fault_tolerance, k8s_monitor\n"
+            f"[-c --config   Config file URL : yaml://<file_path>] E.g. yaml:///etc/cortx/cluster.conf\n"
+            f"[-d --debug   [level]] (if level is not provided, then it considered as 1)"
+            f"\n\t\t0 : No debug.\n\t\t1 : Auto restarts child process if it dia.", file=sys.stderr, flush=True)
 
 def handle_signal(signum, frame):
     """
     Signal handler call beck function, to forward the received signal to driver process
     """
-    sys.stdout.write(f"Signal {signum} at frame {frame} received.")
-    sys.stdout.write(f"Sending {signum} to driver process pid: {driver_process.pid}.")
+    print(f"Signal {signum} at frame {frame} received.", flush=True)
+    print(f"Sending {signum} to driver process pid: {driver_process.pid}.", flush=True)
     driver_process.send_signal(signum)
 
-try:
-    if args.services not in service_entry_mapping:
-        sys.stdout.write('Please provide a valid service name')
-        usage('ha_start')
-        sys.exit(22)
-    # setting Signal handlers for all standered signals
-    signal.signal(signal.SIGABRT, handle_signal)
-    signal.signal(signal.SIGFPE, handle_signal)
-    signal.signal(signal.SIGILL, handle_signal)
-    signal.signal(signal.SIGINT, handle_signal)
-    signal.signal(signal.SIGSEGV, handle_signal)
-    signal.signal(signal.SIGTERM, handle_signal)
-
+def start_driver_process():
+    global driver_process
     driver_process = Popen(['/usr/bin/python3', service_entry_mapping[args.services]], shell=False, stdout=PIPE, stderr=STDOUT) # nosec
+    print(f"The driver process with pid {driver_process.pid}, and args {driver_process.args} started successfully.", flush=True)
+
+def main(argv: list):
+    try:
+        print(str(args))
+        if args.debug not in [0,1]:
+            sys.stderr.write('Please provide a valid debug value.')
+            usage('ha_start')
+            sys.exit(22)
+        if args.services not in service_entry_mapping:
+            print('Please provide a valid service name', flush=True)
+            usage('ha_start')
+            sys.exit(22)
+        # setting Signal handlers for all standered signals
+        signal.signal(signal.SIGABRT, handle_signal)
+        signal.signal(signal.SIGFPE, handle_signal)
+        signal.signal(signal.SIGILL, handle_signal)
+        signal.signal(signal.SIGINT, handle_signal)
+        signal.signal(signal.SIGSEGV, handle_signal)
+        signal.signal(signal.SIGTERM, handle_signal)
+
+        start_driver_process()
+
+        # Poll process.stdout to show stdout live
+        while True:
+            output = driver_process.stdout.readline()
+            if driver_process.poll() is not None:
+                if((args.debug&AUTO_RESTART)!=0):
```

Comment:
I think no need to check != 0 here. & operation will result in 1 or 0. So, if 1 or if 0 can work here.

Reply:
here I am performing a bitwise operation.
0x01 is auto_restart
0x10 is enable debug logging (in future)
0x11 is both (in future)
Here no matter which value is args.debug have I am checking if it has last bin enable i.e. auto restart

Reply:
Right. But, let's see combinations!
auto_restart is always 1. debug can be 1 or 0.
1 & 0: 0 (No auto restart, no debug)
1 & 1: 1 (Debug and auto restart)
Anything I am missing?

Reply:
(in future)
10 & 1: 0 enable debug log but no restart
11 & 1: 1 this value told us to enable auto restart no matter what other things are also required like enabling debug log or more options can be here. like (101 & 1: 1 and 110 & 1: 0 and 100 & 1: 0)

Reply:
Discussed with Mukhtar. Cleared the confusion. Thanks!